### DISCUSSION TITLE
When to use Express/Apollo over just Apollor-Server, what are the use cases?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [apollographql/apollo-server#4369] [General]
Is Apollo + WebFramework only used if we need to build also a UI?

**Comment by abernix** [comment]
The `apollo-server` package is a great _getting started_ package, but if you need to do more advanced customization (or most notably, if you need access to the Express application itself, you'll want to move toward `apollo-server-express` (or any of the other integrations: e.g., Koa, Fastify, Lambda, etc.).

It's common to start with `apollo-server` and then switch to `apollo-server-express` when and if the need arises.  It's generally just a few (additional) lines of code difference that involves calling `applyMiddleware` on an existing application (in the case of `express`, `app` would be the result of `express()`).

We touch on this a bit in the [integration docs](https://www.apollographql.com/docs/apollo-server/integrations/middleware/).
